numpy/core/setup.py

```diff
@@ -605,6 +606,12 @@ def generate_api(ext, build_dir):
     config.add_include_dirs(join('src', 'umath'))
     config.add_include_dirs(join('src', 'npysort'))
 
+    # Need to add location of npy_math_internal.h, generated from src/npymath/npy_math_internal.h.src
+    npymath_build_dir = join('build', 'src.%s-%s.%s' % (sysconfig.get_platform(), sys.version_info[0], sys.version_info[1]),
```

Comment:
It is not an uncommon "bug", clang reports itself as gcc 4.2 but does not support everything gcc does.
numpy is built on a lot of odd platforms and compilers, but I also prefer that these are sorted out after the mainstream platforms have the best code possible.